src/charm.py

```diff
@@ -120,15 +121,15 @@ def render_config(self) -> None:
         Sources for the configuration are charm config options and data from relation
         with exporter charms.
         """
-        config = {
+        config: dict = {
             "settings": {},
             "juju_controller": {},
             "targets": [],
         }
 
         customer = self.config.get("customer")
         site = self.config.get("site")
-        ca_cert = b64decode(self.config.get("juju_ca_cert")).decode("UTF-8")
+        ca_cert = b64decode(self.config.get("juju_ca_cert")).decode("UTF-8")  # type: ignore
```

Comment:
Why are these two `type: ignore` statements necessary? I can't figure out what rule they are breaking

Reply:
I agree with Martin, if it is absolute necessity, I would like to see comment.

Reply:
When I remove the `ignore`, this shows up:
```
src/charm.py:132: error: Argument 1 to "b64decode" has incompatible type "Optional[str]"; expected "Union[str, Union[bytes, Union[bytearray, memoryview, array[Any], mmap, _CData, PickleBuffer]]]"  [arg-type]
src/charm.py:142: error: Item "None" of "Optional[List[Relation]]" has no attribute "__iter__" (not iterable)  [union-attr]
Found 2 errors in 1 file (checked 1 source file)
lint: exit 1 (3.82 seconds) /home/cocenza/Documents/canonical/charm-software-inventory-collector> mypy --install-types --non-interactive . pid=39640
  lint: FAIL code 1 (26.65=setup[19.47]+cmd[0.18,3.17,3.82] seconds)
  evaluation failed :( (26.69 seconds)
```
This is from the ops library that we don't have control of it,

Reply:
I do not agree that you have no controller, the error is due the fact that `self.config.get("juju_ca_cert")` can return `None` and you are not handling it.